Userland/Libraries/LibSemVer/SemVer.cpp

```diff
@@ -0,0 +1,290 @@
+    for (size_t i = 0; i < min(prerelease_parts().size(), other.prerelease_parts().size()); ++i) {
+        auto const this_numerical_identifier = m_prerelease_identifiers[i].to_number<u32>();
+        auto const other_numerical_identifier = other.m_prerelease_identifiers[i].to_number<u32>();
+
+        // 1. Identifiers consisting of only digits are compared numerically.
+        if (this_numerical_identifier.has_value() && other_numerical_identifier.has_value()) {
+            auto const this_value = this_numerical_identifier.value();
+            auto const other_value = other_numerical_identifier.value();
+
+            if (this_value == other_value) {
+                continue;
+            }
+            return this_value > other_value;
+        }
+
+        // 2. Identifiers with letters or hyphens are compared lexically in ASCII sort order.
+        if (!this_numerical_identifier.has_value() && !other_numerical_identifier.has_value()) {
+            if (m_prerelease_identifiers[i] == other.m_prerelease_identifiers[i]) {
+                continue;
+            }
+            return m_prerelease_identifiers[i] > other.m_prerelease_identifiers[i];
+        }
+
+        // 3. Numeric identifiers always have lower precedence than non-numeric identifiers.
+        if (this_numerical_identifier.has_value() && !other_numerical_identifier.has_value())
+            return false;
+        if (!this_numerical_identifier.has_value() && other_numerical_identifier.has_value())
+            return true;
+    }
+
+    // 4. If all of the preceding identifiers are equal, larger set of pre-release fields has a higher precedence than a smaller set.
+    return m_prerelease_identifiers.size() > other.m_prerelease_identifiers.size();
+}
+
+bool SemVer::satisfies(StringView const& semver_spec) const
+{
+    GenericLexer lexer(semver_spec.trim_whitespace());
+    if (lexer.tell_remaining() == 0)
+        return false;
+
+    auto compare_op = lexer.consume_until([](auto const& ch) { return ch >= '0' && ch <= '9'; });
+
+    auto spec_version = MUST(from_string_view(lexer.consume_all()));
+    if (compare_op.length() == 0)
+        return is_same(spec_version, CompareType::Minor);
+    if (compare_op == "!="sv)
+        return !is_same(spec_version);
+
+    // Lenient compare, tolerance for any prerelease or build metadata.
+    if (compare_op == "="sv)
+        return is_same(spec_version, CompareType::Patch);
+
+    // Exact version string match.
+    if (compare_op == "=="sv)
+        return is_same(spec_version);
+
+    // Current version is greater than spec.
+    if (compare_op == ">"sv)
+        return is_greater_than(spec_version);
+    if (compare_op == "<"sv)
+        return is_lesser_than(spec_version);
+    if (compare_op == ">="sv)
+        return is_same(spec_version) || is_greater_than(spec_version);
+    if (compare_op == "<="sv)
+        return is_same(spec_version) || is_lesser_than(spec_version);
```

Comment:
given the implementation of is_less_than as `!is_same && !greater_than`, this is very inefficient. less equal should just use `!greater_than` directly.

Reply:
Actually that would fail `1.1.2-prerelease+meta<=1.1.2-prerelease+meta` case. Left one is version and right one is the spec

Reply:
I am using `is_same(spec_version) || !is_greater_than(spec_version);` now. That would fix the performance issue.